### ISSUE TITLE
Imcompatible case with ES5 implementation

### ISSUE THREAD

**Issue by koba04** [mweststrate/immer#116]
> In such cases Immer will fallback to an ES5 compatible implementation which works identical, but is a bit slower.

`immer` says the ES5 implementation is compatible with the implementation using Proxy.
But the following test fails when I use ES5 implementation.

```js
const nextState = produce([1,2,3], s => {
  s.pop();
  s.push(100);
})
expect(nextState).toEqual([1,2,100])
// ES5 implmentation 
//    Expected value to equal:
//      [1, 2, 100]
//    Received:
//      [1, 2, 3]
```

Is this an expected behavior?

**Comment by mweststrate** [comment]
Nope. reproduction: https://codesandbox.io/s/vmxmn8omwl

**Comment by mweststrate** [comment]
Released as 1.1.3. Thanks for reporting!

**Comment by koba04** [comment]
Thanks! 👏
